### ISSUE TITLE
Converted Audio sounds the same as Original Audio - ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

### ISSUE THREAD

**Issue by AchillesSeal** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#1207]
Hello. 
I was hoping someone might be able to help me. I would like to start this by saying that I have no experience with coding and I don’t even understand the basics, I have only gotten this far by watching videos on Youtube and following the instructions. The only experience I have is that I managed to get Stable Diffusion to work. 

I have been trying to train a voice in RVC and I have encountered a number of problems, thus far I have been able to find solutions myself, without altering anything, most errors were a result of me not doing something exactly right, but I figured it out in the end. However, I have now encountered an error that occurs after I try to “Convert” during Model Inference.

I load the necessary files of the model, index and song vocals I want to use, I click “Convert” and the program starts working. After about 30 seconds or so, the conversion finishes and it produces the new audio file, however, the audio that is produced is almost identical to the original vocals (with a little artifacting and a little metallic sounding). Furthermore, I can only hear the new audio if I turn the "Search feature ratio" down quite considerably, if I don't turn this parameter down, you can't hear anything, even if you turn "Adjust the volume envelope scaling" to 1.

When I look at the coding, the following error occurs:

2023-09-07 12:04:12 | ERROR | asyncio | Exception in callback _ProactorBasePipeTransport._call_connection_lost(None)
handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)>
Traceback (most recent call last):
  File "C:\Users\steph\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\steph\AppData\Local\Programs\Python\Python310\lib\asyncio\proactor_events.py", line 165, in _call_connection_lost
    self._sock.shutdown(socket.SHUT_RDWR)
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

Like I said before, I’m completely new to all of this sort of stuff and I would greatly appreciate the help because I haven’t got a clue what I’m doing.

Many Thanks

**Comment by lvt1693** [comment]
Did you try using other models  to see if it works? ((you can find a lot of them on huggingface)
- If it works, then there's something wrong with your trained model.
- Else, just get a fresh install :(

**Comment by AchillesSeal** [comment]
> Did you try using other models to see if it works? ((you can find a lot of them on huggingface)
> 
> * If it works, then there's something wrong with your trained model.
> * Else, just get a fresh install :(

I’ve just downloaded a model online of Eminem and given it a test, I got the same error, but different results. This time, the converted audio sounded like Eminem, though it still had quite a bad glitchy-metallic sound with it, just like before with the model I trained. I compared the Eminem model I downloaded with the model that I trained myself and I noticed a number of differences that most likely are adding to the problem.

First, the name of the .pth files in the “logs” folder. The name of the Eminem model’s .pth file shared the name of the subject (Eminem2333333.pth), meanwhile, the model I trained had 2 .pth files, neither had a name even remotely close to the subject (D_2333333.pth and G_2333333.pth), the name of the subject was a friend’s name, so I won’t put it down. When I copy my trained model’s .pth files to the “weights” folder and then use them in the program, I just get “Error” appear on the screen and it won’t let me use “Convert”.

Secondly, the sizes of the index and .pth files. The Eminem model’s index file sizes were 10,453 KB and 62,235 KB (there were 2 index files) and the .pth file was 53,827 KB.

Meanwhile, the model I trained, the “added” index file was only 458 KB (it also had a “trained” index file of only 10 KB) and the 2 .pth file sizes in the “logs” folder were massive, 837,038 KB and 427,914 KB (D_2333333.pth and G_2333333.pth respectively).

As a note, the Eminem Model was trained with 800 epochs while mine was trained with 1,000 epochs. I don’t know if that’s of any relevance.

Lastly, the .pth files of my trained model are different in the “weights” folder. Here, they do have the subject’s name and the number of epochs trained (FriendName_e1000_s1000.pth) with a file size of 53,937 KB, almost the same as the Eminem models .pth file size. Unlike the files of my trained model from the “logs” folder, RVC accepts this .pth file, but like I said before, the converted audio is the same as the song vocals. 

Like I said in my previous message, I’m completely new to this and so, I know very little about this, but it seems to me the problem is with my model’s index file size, though I don’t know what would have caused it to be so small. Also, I still don’t know what’s causing the glitchy-metallic sound of the converted audio, even with the Eminem model I downloaded, my guess would be the error that keeps appearing.

Thanks again for any help.

**Comment by AchillesSeal** [comment]
An update for my ongoings with this. I have finally managed to get my trained model to sound like my subject

After my last message, I looked at how I put in the Dataset for the training audio and found that I hadn't preprocessed the data properly. I learnt that if you don't put the path to the folder containing your dataset in quotation marks, "like this", the program doesn't preprocess the data, and that's why the index files were so small, thereby unable to replicate my subject's voice.

However, unfortunately, I am still suffering from the following error message whenever I try to use Model Inference:

2023-09-09 09:47:10 | ERROR | asyncio | Exception in callback _ProactorBasePipeTransport._call_connection_lost(None)
handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)>
Traceback (most recent call last):
  File "C:\Users\steph\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\steph\AppData\Local\Programs\Python\Python310\lib\asyncio\proactor_events.py", line 165, in _call_connection_lost
    self._sock.shutdown(socket.SHUT_RDWR)
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

While the audio now sounds like my subject, it is very glitchy-metallic sounding and turning the "Search feature ratio" to 0 only lessens this a little bit. The main thing I noticed when I click "Convert", the processing finishes after a few seconds and ends at roughly "processing | 27.4 / 4283.3", as shown just above where the converted audio appears. This implies to me that it's not being processed properly and is either ending prematurely or being aborted altogether, which is consistent with the error message.

As always, thanks to anyone who is able to help.

**Comment by elfuriksson** [comment]
Hi
I have the exact same problem as you describe. Seems like this thread's closed as completed but did you find any solution? Any help's appreciated

**Comment by AchillesSeal** [comment]
> Hi I have the exact same problem as you describe. Seems like this thread's closed as completed but did you find any solution to the "ConnectionResetError: [WinError 10054]" and the glitchy sound? Any help's appreciated

Hello,
In the end, I ignored the Error as I managed to get it working regardless of it, I don't know why the Error appears. I found what was causing the glitchy sound however.

Quite simply, the audio file of the vocals that was generated using the software wasn't of very good quality. To resolve the issue, I used a free online service that was better in 2 regards:
1) The quality of the audio was better which drastically reduced the glitchy sound when converted to your model
2) Not only does it separate the music from the vocals. It also separates the backing vocals

Here is a link to the site I used to separate the music and vocals:
https://www.audiostrip.co.uk/isolate 

I hope all of this information helps.

**Comment by elfuriksson** [comment]
Ok. Thanks for the answer. 
I just can't see that my audio for the training is that bad though. I've got around 20 min of, in my ears, clean edited vocals without noice, reverb, other instruments or vocals. There are some reverb in the timbre at places but nothing that rings after the tone's finnished. The vocals got quite some compression on the recording I've isolated it from though. 

I've run in to your initial problem as well when testing with some isolated vocals from another track. Then there wasn't nearly any difference from the original vocals, other then some kind of digital sound going on. Wonder if it might be something wrong with my audio files in the dataset. I made like 150 10 second wav-files before I even started using RVC. Because of some tutorial that said so. 

I'll take your advice and create a new dataset. First with the same recordings but in a different way, then a completely new one with "better" audio if the problem continues.

Does it still take like a couple of seconds for the conversion before the error? Even when it sounds good? Seems strange to me.
What version of RVC do you use?

I'm very thankful for yor time and help.

**Comment by AchillesSeal** [comment]
> Ok. Thanks for the answer. I just can't see that my audio for the training is that bad though. I've got around 20 min of, in my ears, clean edited vocals without noice, reverb, other instruments or vocals. There are some reverb in the timbre at places but nothing that rings after the tone's finnished. The vocals got quite some compression on the recording I've isolated it from though.
> 
> I've run in to your initial problem as well when testing with some isolated vocals from another track. Then there wasn't nearly any difference from the original vocals, other then some kind of digital sound going on. Wonder if it might be something wrong with my audio files in the dataset. I made like 150 10 second wav-files before I even started using RVC. Because of some tutorial that said so.
> 
> I'll take your advice and create a new dataset. First with the same recordings but in a different way, then a completely new one with "better" audio if the problem continues.
> 
> Does it still take like a couple of seconds for the conversion before the error? Even when it sounds good? Seems strange to me. What version of RVC do you use?
> 
> I'm very thankful for yor time and help.

I think there may be a bit of confusion. When I was referring to the audio for the vocals, I was referring to the song/piece you wanted your trained model to replicate. For example, I tried to make someone sing "Thriller" by Michael Jackson.

I initially used the "Vocals/Accompaniment Separation & Reverberation Removal" in the software to separate the vocals and music for "Thriller", but THIS file was of poor quality. I wasn't referring to the audio used to train my model. So when I put the poor quality song vocals of "Thriller" generated by "Vocals/Accompaniment Separation & Reverberation Removal" into "Model Inference", my model replicated the poor quality song vocals. I used the site I linked previously to give me better quality audio for "Thriller".

From what you have said, I don't think the quality of the audio of your subject for the dataset is the problem, but rather, the audio that you're trying to make them mimic.

Regarding training my model, I just want to make some notes that I found that ultimately got it to work fo me:
1) In training, in "Enter the path of the training folder", the path to the folder MUST be in quotation marks. For example "C:\Users\me\Folder1\Folder2\Folder3"

If the path is not in quotation marks, it cannot find your dataset to train your model and "Process Data" will fail to generate the correct folders.

2) I trained my model at "200 epochs" and batch size of 4. With my dataset being a total of 50 minutes of my subject, it took 12 hours to train my model. If your training is completed in a few minutes, something is wrong

Finally, regarding conversion, the conversion process was not long at all. It varied between a few seconds to maybe a minute or two if my computer suddenly decided it wanted to be slow. Also, and I've read other people have noticed this, you still get the Error even when the conversion was a success, this happened with me too. This has led people to suspect that it's simply a bug and I just ignored it because the conversion was a success and my model was now singing "Thriller". I believe that I am using the most recent version of RVC.

One last thing, my model didn't like high notes and still sounded a LITTLE bit scratchy when it tried to hit the high notes, I managed to smooth this out a bit when I was editing the song and putting things back together. When it was singing "normal" notes, like within talking range, it sounded fine.
